examples/junit5-arquillian-simple-websockets/src/test/java/org/superbiz/websockets/WebSocketResourceTest.java

```diff
@@ -0,0 +1,95 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.superbiz.websockets;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit5.ArquillianExtension;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import javax.websocket.*;
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@RunAsClient
+@ExtendWith(ArquillianExtension.class)
+public class WebSocketResourceTest {
+
+    private static final int PORT = 8081;
+
+    @ArquillianResource()
+    private URL base;
+
+    @Deployment(testable = false)
+    public static final WebArchive app() {
+        return ShrinkWrap.create(WebArchive.class, "demo.war")
+                .addClasses(WebSocketResource.class)
+                .addAsWebInfResource(new File("src/main/webapp/WEB-INF/web.xml"), "web.xml")
+                .addAsWebInfResource(new File("src/main/resources/META-INF/beans.xml"), "beans.xml");
+    }
+
+    @Test
+    public void testConnectAndReceiveMessage() throws Exception {
+        Session session = connectToServer(MyWebSocketClientObject.class);
+        assertNotNull(session);
+
+        assertTrue(MyWebSocketClientObject.latch.await(2, TimeUnit.SECONDS));
+        assertEquals("Successfully opened session", MyWebSocketClientObject.response);
+
+        session.close();
+    }
+
+    @Test
+    public void testConnectAndSendPayload() throws Exception {
+
+        String payload = "I am the payload sent to this resource.";
+        Session session = connectToServer(MyWebSocketClientObject.class);
+        assertNotNull(session);
+
+        assertTrue(MyWebSocketClientObject.latch.await(2, TimeUnit.SECONDS));
+        assertEquals("Successfully opened session", MyWebSocketClientObject.response);
+
+        session.getBasicRemote().sendText(payload);
+
+        assertTrue(MyWebSocketClientObject.payloadLatch.await(2, TimeUnit.SECONDS));
+        assertEquals("Received: " + payload, MyWebSocketClientObject.response);
+
+        session.close();
+    }
+
+    /**
+     * Method used to supply connection to the server by passing the naming of
+     * the websocket endpoint
+     */
+    public Session connectToServer(Class<?> endpoint) throws DeploymentException, IOException, URISyntaxException {
+        WebSocketContainer container = ContainerProvider.getWebSocketContainer();
+        assertNotNull(container);
+        return container.connectToServer(endpoint, new URI("ws", base.getUserInfo(), "localhost", PORT,"/api/socket",null, null));
```

Comment:
Once arquillian dinamicaly allocate an available port (check my other comment) you can obtain the webTarget with `base.toExternalForm()`
Excample
https://github.com/apache/tomee/blob/master/examples/mp-custom-healthcheck/src/test/java/org/superbiz/test/WeatherServiceTest.java#L68

Reply:
Fixed.